src/core/shape_inference/include/stft_shape_inference.hpp

```diff
@@ -25,10 +25,11 @@ std::vector<TRShape> shape_infer(const STFT* op,
     const auto& frame_size_shape = input_shapes[2];
     const auto& frame_step_shape = input_shapes[3];
 
+    const auto signal_shape_rank = signal_shape.rank();
     NODE_SHAPE_INFER_CHECK(op,
                            input_shapes,
-                           signal_shape.rank().compatible(2),
-                           "The shape of signal must be 2D [batch, signal_size].");
+                           signal_shape_rank.compatible(1) || signal_shape_rank.compatible(2),
```

Comment:
```suggestion
                           is_rank_compatible_any_of(signal_shape_rank, {1, 2}),
```
?

Reply:
As the logic stays the same and it's a just suggestion to use helper function, I will consider to apply it as a small follow up refactor. Thanks!